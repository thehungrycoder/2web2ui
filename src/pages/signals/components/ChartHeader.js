import React, { Fragment } from 'react';
import { Tooltip } from '@sparkpost/matchbox';
import { InfoOutline } from '@sparkpost/matchbox-icons';
import { formatDate } from 'src/helpers/date';
import styles from './ChartHeader.module.scss';

const ChartHeader = ({ date, title, primaryArea, hideLine, tooltipContent, padding = '' }) => (
  <Fragment>
    <div className={styles.ChartHeader} style={{ padding }}>
      <h6 className={styles.Title}>{title}{date && ` â€“ ${formatDate(date)}`}</h6>
      {tooltipContent && (
        <div className={styles.Tooltip}>
          <Tooltip dark horizontalOffset='-1rem' content={tooltipContent}>
            <InfoOutline className={styles.TooltipIcon} size={17} />
          </Tooltip>
        </div>
      )}
      {primaryArea && <div className={styles.PrimaryArea}>{primaryArea}</div>}
    </div>
    {!hideLine && <hr className={styles.Line} />}
  </Fragment>
);

export default ChartHeader;
