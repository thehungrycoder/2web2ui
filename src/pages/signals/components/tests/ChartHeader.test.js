import { shallow } from 'enzyme';
import React from 'react';
import ChartHeader from '../ChartHeader';

describe('Signals ChartHeader Component', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      title: 'Foo',
      date: '2018-02-15'
    };
    wrapper = shallow(<ChartHeader {...props}/>);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders primaryArea correctly', () => {
    wrapper.setProps({ primaryArea: <div>react</div> });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders tooltip content correctly', () => {
    wrapper.setProps({ tooltipContent: 'tooltip content' });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders padding correctly', () => {
    wrapper.setProps({ padding: '0 0 1rem 0' });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders without date', () => {
    wrapper.setProps({ date: null });
    expect(wrapper).toMatchSnapshot();
  });
});
