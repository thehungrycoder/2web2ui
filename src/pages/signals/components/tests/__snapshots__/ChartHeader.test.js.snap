// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signals ChartHeader Component renders correctly 1`] = `
<Fragment>
  <div
    className="ChartHeader"
    style={
      Object {
        "padding": "",
      }
    }
  >
    <h6
      className="Title"
    >
      Foo
       – Feb 15 2018
    </h6>
  </div>
  <hr
    className="Line"
  />
</Fragment>
`;

exports[`Signals ChartHeader Component renders padding correctly 1`] = `
<Fragment>
  <div
    className="ChartHeader"
    style={
      Object {
        "padding": "0 0 1rem 0",
      }
    }
  >
    <h6
      className="Title"
    >
      Foo
       – Feb 15 2018
    </h6>
  </div>
  <hr
    className="Line"
  />
</Fragment>
`;

exports[`Signals ChartHeader Component renders primaryArea correctly 1`] = `
<Fragment>
  <div
    className="ChartHeader"
    style={
      Object {
        "padding": "",
      }
    }
  >
    <h6
      className="Title"
    >
      Foo
       – Feb 15 2018
    </h6>
    <div
      className="PrimaryArea"
    >
      <div>
        react
      </div>
    </div>
  </div>
  <hr
    className="Line"
  />
</Fragment>
`;

exports[`Signals ChartHeader Component renders tooltip content correctly 1`] = `
<Fragment>
  <div
    className="ChartHeader"
    style={
      Object {
        "padding": "",
      }
    }
  >
    <h6
      className="Title"
    >
      Foo
       – Feb 15 2018
    </h6>
    <div
      className="Tooltip"
    >
      <Tooltip
        bottom={true}
        content="tooltip content"
        dark={true}
        forcePosition={false}
        horizontalOffset="-1rem"
        preferredDirection={
          Object {
            "bottom": true,
            "left": false,
            "right": true,
            "top": false,
          }
        }
        right={true}
      >
        <InfoOutline
          className="TooltipIcon"
          size={17}
        />
      </Tooltip>
    </div>
  </div>
  <hr
    className="Line"
  />
</Fragment>
`;

exports[`Signals ChartHeader Component renders without date 1`] = `
<Fragment>
  <div
    className="ChartHeader"
    style={
      Object {
        "padding": "",
      }
    }
  >
    <h6
      className="Title"
    >
      Foo
    </h6>
  </div>
  <hr
    className="Line"
  />
</Fragment>
`;
